### PR TITLE
Fix mutex locking around ifname cache

### LINKED ISSUE #8842: IfName plugin causes Telegraf to crash
### Relevant telegraf.conf:

```toml
# Lookup interface name for qos_portqueue table
[[processors.ifname]]
  namepass = ["qos_portqueue"]

  ## Name of tag holding the interface number
  tag = "InterfaceIndex"

  ## Name of output tag where service name will be added
  # dest = "ifName"

  ## Name of tag of the SNMP agent to request the interface name from
  agent = "agent_host"

  ## Timeout for each request.
  timeout = "5s"

  ## SNMP version; can be 1, 2, or 3.
  version = 2

  ## SNMP community string.
  community = "public"

  ## Number of retries to attempt.
  retries = 3

  ## The GETBULK max-repetitions parameter.
  max_repetitions = 100

  ## cache_ttl is the amount of time interface names are cached for a
  ## given agent.  After this period elapses if names are needed they
  ## will be retrieved again.
  cache_ttl = "15m"

```

### System info:

Telegraf 1.17.2 (git: HEAD 74011e22) on RHEL 7.4

### Steps to reproduce:

1. Configure ifname plugin
2. Run telegraf

### Expected behavior:

Telegraf not to crash when using ifname processor.

### Actual behavior:

Telegraf crashes sometimes when the ifname cache refreshes. So the time telegraf can run continuously is a multiple of the configured `cache_ttl`.

### Additional info:

Stack trace: https://pastebin.com/ty9YxRHb
Trace 2: https://pastebin.com/JCfAhE2W

### PR DESCRIPTION
**Pull request by reimda**
fixes #8842

### CODE REVIEW

**Comment by reimda** [comment]
Here are the packages from CI for this PR in case they make testing easier for you.  Thanks!

<ul class="css-18t69f4"><li class="css-7kuxj3"><a href="https://78700-33258973-gh.circle-artifacts.com/0/build/dist/telegraf-1.18.0%7E7773765e-0.aarch64.rpm" class="css-ltpbf2">build/dist/telegraf-1.18.0~7773765e-0.aarch64.rpm</a></li><li class="css-7kuxj3"><a href="https://78700-33258973-gh.circle-artifacts.com/0/build/dist/telegraf-1.18.0%7E7773765e-0.armel.rpm" class="css-ltpbf2">build/dist/telegraf-1.18.0~7773765e-0.armel.rpm</a></li><li class="css-7kuxj3"><a href="https://78700-33258973-gh.circle-artifacts.com/0/build/dist/telegraf-1.18.0%7E7773765e-0.armv6hl.rpm" class="css-ltpbf2">build/dist/telegraf-1.18.0~7773765e-0.armv6hl.rpm</a></li><li class="css-7kuxj3"><a href="https://78700-33258973-gh.circle-artifacts.com/0/build/dist/telegraf-1.18.0%7E7773765e-0.i386.rpm" class="css-ltpbf2">build/dist/telegraf-1.18.0~7773765e-0.i386.rpm</a></li><li class="css-7kuxj3"><a href="https://78700-33258973-gh.circle-artifacts.com/0/build/dist/telegraf-1.18.0%7E7773765e-0.ppc64le.rpm" class="css-ltpbf2">build/dist/telegraf-1.18.0~7773765e-0.ppc64le.rpm</a></li><li class="css-7kuxj3"><a href="https://78700-33258973-gh.circle-artifacts.com/0/build/dist/telegraf-1.18.0%7E7773765e-0.s390x.rpm" class="css-ltpbf2">build/dist/telegraf-1.18.0~7773765e-0.s390x.rpm</a></li><li class="css-7kuxj3"><a href="https://78700-33258973-gh.circle-artifacts.com/0/build/dist/telegraf-1.18.0%7E7773765e-0.x86_64.rpm" class="css-ltpbf2">build/dist/telegraf-1.18.0~7773765e-0.x86_64.rpm</a></li><li class="css-7kuxj3"><a href="https://78700-33258973-gh.circle-artifacts.com/0/build/dist/telegraf-1.18.0%7E7773765e_darwin_amd64.tar.gz" class="css-ltpbf2">build/dist/telegraf-1.18.0~7773765e_darwin_amd64.tar.gz</a></li><li class="css-7kuxj3"><a href="https://78700-33258973-gh.circle-artifacts.com/0/build/dist/telegraf-1.18.0%7E7773765e_freebsd_amd64.tar.gz" class="css-ltpbf2">build/dist/telegraf-1.18.0~7773765e_freebsd_amd64.tar.gz</a></li><li class="css-7kuxj3"><a href="https://78700-33258973-gh.circle-artifacts.com/0/build/dist/telegraf-1.18.0%7E7773765e_freebsd_i386.tar.gz" class="css-ltpbf2">build/dist/telegraf-1.18.0~7773765e_freebsd_i386.tar.gz</a></li><li class="css-7kuxj3"><a href="https://78700-33258973-gh.circle-artifacts.com/0/build/dist/telegraf-1.18.0%7E7773765e_linux_amd64.tar.gz" class="css-ltpbf2">build/dist/telegraf-1.18.0~7773765e_linux_amd64.tar.gz</a></li><li class="css-7kuxj3"><a href="https://78700-33258973-gh.circle-artifacts.com/0/build/dist/telegraf-1.18.0%7E7773765e_linux_arm64.tar.gz" class="css-ltpbf2">build/dist/telegraf-1.18.0~7773765e_linux_arm64.tar.gz</a></li><li class="css-7kuxj3"><a href="https://78700-33258973-gh.circle-artifacts.com/0/build/dist/telegraf-1.18.0%7E7773765e_linux_armel.tar.gz" class="css-ltpbf2">build/dist/telegraf-1.18.0~7773765e_linux_armel.tar.gz</a></li><li class="css-7kuxj3"><a href="https://78700-33258973-gh.circle-artifacts.com/0/build/dist/telegraf-1.18.0%7E7773765e_linux_armhf.tar.gz" class="css-ltpbf2">build/dist/telegraf-1.18.0~7773765e_linux_armhf.tar.gz</a></li><li class="css-7kuxj3"><a href="https://78700-33258973-gh.circle-artifacts.com/0/build/dist/telegraf-1.18.0%7E7773765e_linux_i386.tar.gz" class="css-ltpbf2">build/dist/telegraf-1.18.0~7773765e_linux_i386.tar.gz</a></li><li class="css-7kuxj3"><a href="https://78700-33258973-gh.circle-artifacts.com/0/build/dist/telegraf-1.18.0%7E7773765e_linux_mips.tar.gz" class="css-ltpbf2">build/dist/telegraf-1.18.0~7773765e_linux_mips.tar.gz</a></li><li class="css-7kuxj3"><a href="https://78700-33258973-gh.circle-artifacts.com/0/build/dist/telegraf-1.18.0%7E7773765e_linux_mipsel.tar.gz" class="css-ltpbf2">build/dist/telegraf-1.18.0~7773765e_linux_mipsel.tar.gz</a></li><li class="css-7kuxj3"><a href="https://78700-33258973-gh.circle-artifacts.com/0/build/dist/telegraf-1.18.0%7E7773765e_linux_ppc64le.tar.gz" class="css-ltpbf2">build/dist/telegraf-1.18.0~7773765e_linux_ppc64le.tar.gz</a></li><li class="css-7kuxj3"><a href="https://78700-33258973-gh.circle-artifacts.com/0/build/dist/telegraf-1.18.0%7E7773765e_linux_s390x.tar.gz" class="css-ltpbf2">build/dist/telegraf-1.18.0~7773765e_linux_s390x.tar.gz</a></li><li class="css-7kuxj3"><a href="https://78700-33258973-gh.circle-artifacts.com/0/build/dist/telegraf-1.18.0%7E7773765e_static_linux_amd64.tar.gz" class="css-ltpbf2">build/dist/telegraf-1.18.0~7773765e_static_linux_amd64.tar.gz</a></li><li class="css-7kuxj3"><a href="https://78700-33258973-gh.circle-artifacts.com/0/build/dist/telegraf-1.18.0%7E7773765e_windows_amd64.zip" class="css-ltpbf2">build/dist/telegraf-1.18.0~7773765e_windows_amd64.zip</a></li><li class="css-7kuxj3"><a href="https://78700-33258973-gh.circle-artifacts.com/0/build/dist/telegraf-1.18.0%7E7773765e_windows_i386.zip" class="css-ltpbf2">build/dist/telegraf-1.18.0~7773765e_windows_i386.zip</a></li><li class="css-7kuxj3"><a href="https://78700-33258973-gh.circle-artifacts.com/0/build/dist/telegraf_1.18.0%7E7773765e-0_amd64.deb" class="css-ltpbf2">build/dist/telegraf_1.18.0~7773765e-0_amd64.deb</a></li><li class="css-7kuxj3"><a href="https://78700-33258973-gh.circle-artifacts.com/0/build/dist/telegraf_1.18.0%7E7773765e-0_arm64.deb" class="css-ltpbf2">build/dist/telegraf_1.18.0~7773765e-0_arm64.deb</a></li><li class="css-7kuxj3"><a href="https://78700-33258973-gh.circle-artifacts.com/0/build/dist/telegraf_1.18.0%7E7773765e-0_armel.deb" class="css-ltpbf2">build/dist/telegraf_1.18.0~7773765e-0_armel.deb</a></li><li class="css-7kuxj3"><a href="https://78700-33258973-gh.circle-artifacts.com/0/build/dist/telegraf_1.18.0%7E7773765e-0_armhf.deb" class="css-ltpbf2">build/dist/telegraf_1.18.0~7773765e-0_armhf.deb</a></li><li class="css-7kuxj3"><a href="https://78700-33258973-gh.circle-artifacts.com/0/build/dist/telegraf_1.18.0%7E7773765e-0_i386.deb" class="css-ltpbf2">build/dist/telegraf_1.18.0~7773765e-0_i386.deb</a></li><li class="css-7kuxj3"><a href="https://78700-33258973-gh.circle-artifacts.com/0/build/dist/telegraf_1.18.0%7E7773765e-0_mips.deb" class="css-ltpbf2">build/dist/telegraf_1.18.0~7773765e-0_mips.deb</a></li><li class="css-7kuxj3"><a href="https://78700-33258973-gh.circle-artifacts.com/0/build/dist/telegraf_1.18.0%7E7773765e-0_mipsel.deb" class="css-ltpbf2">build/dist/telegraf_1.18.0~7773765e-0_mipsel.deb</a></li><li class="css-7kuxj3"><a href="https://78700-33258973-gh.circle-artifacts.com/0/build/dist/telegraf_1.18.0%7E7773765e-0_ppc64el.deb" class="css-ltpbf2">build/dist/telegraf_1.18.0~7773765e-0_ppc64el.deb</a></li><li class="css-7kuxj3"><a href="https://78700-33258973-gh.circle-artifacts.com/0/build/dist/telegraf_1.18.0%7E7773765e-0_s390x.deb" class="css-ltpbf2">build/dist/telegraf_1.18.0~7773765e-0_s390x.deb</a></li></ul>

**Review by Hipska — APPROVED** [review]

File: plugins/processors/ifname/ifname.go [hunk 1]
```diff
@@ -241,60 +241,61 @@ func (d *IfName) Stop() error {
...
-		d.rwLock.Unlock()
 
+		d.lock.Unlock()
 		return nil, 0, fmt.Errorf("getting remote table: %w", err)
 	}
 
-	// Cache it, then signal any other waiting requests for this agent
-	// and clean up
-	d.rwLock.Lock()
+	// snmp success.  Cache response, then signal any other waiting
```
Comment (Jagularr) [inline plugins/processors/ifname/ifname.go:292]: nit: any reason for having the two spaces after the period?

**Review by Jagularr — APPROVED** [review]

File: plugins/processors/ifname/ifname.go (hunk 1 above)
Comment (reimda) [reply plugins/processors/ifname/ifname.go:292]: When I learned to type that's what they taught in school. I'm aware that single space is more common now but it's a habit.

See https://en.wikipedia.org/wiki/Sentence_spacing

**Comment by sspaink** [comment]
!signed-cla

**Pull request merged**